### PR TITLE
Update determine runner workflow to use large runners if the branch name matches a certain format

### PR DESCRIPTION
**Pull request by rashidnhm**
**Context:**
This PR is being opened under sc-66351. It introduces a new change to CI ahead of release where a local branch (not forked) that matches the regex pattern `vX.Y.Z-rcN` (The `N` after rc being optional) is auto ported over to the large runners.

Though this may provide a minor bump to the runtimes of the workflows themselves, the bigger objective (right now at least) is to not wait long times to get a runner.

**Description of the Change:**
The change is a minor one within the `determine-runner` workflow. It now checks for both a label present on a pr OR if the pr branch names matches the rc branching format above.

**Benefits:**
Faster queue times for rc branches.

**Possible Drawbacks:**
Though the large runners and standard runners should have parity in theory. We do not live in a perfect world. There is a none-zero chance that workflows might break. But we can test that out as part of this PR.

**Related GitHub Issues:**
None.

### CODE REVIEW

**Comment by AmintorDusko** [comment]
Hey @rashidnhm, how are we testing this change?

**Comment by rashidnhm** [comment]
> Hey @rashidnhm, how are we testing this change?

Hey @AmintorDusko, we can test the large runners themselves by adding the urgent label to this PR.

To test the auto on-board, I have tested the expression locally against the past rc branch names, we can also create an rc branch name and test it (but I didn't want to do that right now to cause confusion :sweat_smile: )

**Review by maliasadi — APPROVED** [review]
Thanks @rashidnhm!

**Review by vincentmr — CHANGES_REQUESTED** [review]
Thanks @rashidnhm . I'm trusting this works. Please don't forget to update the CHANGELOG with a description and your name, and then I'll approve.

**Comment by AmintorDusko** [comment]
Thank you, @rashidnhm!

**Comment by rashidnhm** [comment]
> Thanks @rashidnhm . I'm trusting this works. Please don't forget to update the CHANGELOG with a description and your name, and then I'll approve.

Thank you for that @vincentmr. If a bug does arise in the logic, we can definitely fall forwards and fix the issue. We also still retain the ability to port any PR to using large runners with the `urgent` label (this includes rc branches).

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,8 @@
 
 ### Improvements
 
+* Release candidate branches are automatically switched over to using the new large GitHub runner pool [(#769)](https://github.com/PennyLaneAI/pennylane-lightning/pull/769)
```
Comment (AmintorDusko) [inline .github/CHANGELOG.md:38]:
```suggestion
* Release candidate branches are automatically switched over to using the new large GitHub runner pool.
   [(#769)](https://github.com/PennyLaneAI/pennylane-lightning/pull/769)
```

**Review by AmintorDusko — APPROVED** [review]
Nice job! Only one more suggestion. Thanks!

**Review by vincentmr — APPROVED** [review]
LGTM, thanks @rashidnhm .

**Pull request merged**